src/System.Text.Json/tests/Serialization/Value.ReadTests.ConcurrentCollections.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Concurrent;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static partial class ValueTests
+    {
+        [Fact]
+        public static void Read_ConcurrentCollection()
+        {
+            ConcurrentDictionary<string, string> cd = JsonSerializer.Deserialize<ConcurrentDictionary<string, string>>(@"{""key"":""value""}");
+            Assert.Equal(1, cd.Count);
+            Assert.Equal("value", cd["key"]);
+
+            ConcurrentQueue<string> qc = JsonSerializer.Deserialize<ConcurrentQueue<string>>(@"[""1""]");
+            Assert.Equal(1, qc.Count);
+            qc.TryPeek(out string val);
+            Assert.Equal("1", val);
+
+            ConcurrentStack<string> qs = JsonSerializer.Deserialize<ConcurrentStack<string>>(@"[""1""]");
+            Assert.Equal(1, qs.Count);
+            qc.TryPeek(out val);
+            Assert.Equal("1", val);
+        }
+
+        [Fact]
+        public static void Read_ConcurrentCollection_Throws()
+        {
+            // Not supported. Not IList, and we don't detect the add method for this collection.
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<BlockingCollection<string>>(@"[""1""]"));
+
+            // Not supported. Not IList, and we don't detect the add method for this collection.
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<ConcurrentBag<string>>(@"[""1""]"));
```

Comment:
Why isn't `ConcurrentBag<T>` supported but `ConcurrentQueue<T>` and `ConcurrentStack<T>` are?  All three implement the same two interfaces: `IProducerConsumerCollection<T>` and `IReadOnlyCollection<T>`.
https://github.com/dotnet/corefx/blob/007f4456374c582ebc7a7a678072935a2560eee5/src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentStack.cs#L40
https://github.com/dotnet/corefx/blob/007f4456374c582ebc7a7a678072935a2560eee5/src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentBag.cs#L33
https://github.com/dotnet/corefx/blob/007f4456374c582ebc7a7a678072935a2560eee5/src/Common/src/CoreLib/System/Collections/Concurrent/ConcurrentQueue.cs#L22